File: src/TerribleDev.Blog.Web/Models/Post.cs
﻿using Microsoft.AspNetCore.Html;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Diagnostics.CodeAnalysis;

namespace TerribleDev.Blog.Web.Models
{
    [DebuggerDisplay("{Title}")]
    public class Post : IPost
    {
        public string CanonicalUrl { get; set; }
        public string UrlWithoutPath { get; set; }
        public string RelativeUrl { get; set; }
        public string Title { get; set; }
        public DateTime PublishDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public IList<string> tags { get; set; }
        public IPostContent Content { get; set; }

        public bool isLanding { get; set; } = false;

        public string ThumbnailImage { get; set; }
    }
}

File: src/TerribleDev.Blog.Core/Models/Post.fs
namespace TerribleDev.Blog.Core.Models
open System;
open Microsoft.AspNetCore.Html;
open System.Collections.Generic

type Post =
    {
        Url: string;
        Title: string;
        PublishDate: DateTime;
        Content: HtmlString;
        Summary: HtmlString; 
        ContentPlain: string; 
        SummaryPlain: string; 
        SummaryPlainShort: string; 
        tags : IList<string>
        Images: IList<string>
    }